OpenProjectImporter/script.js

```diff
@@ -0,0 +1,386 @@
+            const projectURL = workPackage._links.project.href;
+            projectIDs.add(this.extractProjectID(projectURL));
+        }
+        this.projects = this.apiClient.loadProjects(projectIDs);
+
+        // Load project categories
+        const categoryIDs = new Set();
+        for (let projectID in this.projects) {
+            const project = this.projects[projectID];
+            const categoryID = this.getCategoryIDFromProject(project);
+            if (categoryID) {
+                categoryIDs.add(categoryID);
+            }
+        }
+        this.categories = Array.from(categoryIDs);
+
+        // Create projects and work packages
+        this.processData();
+    }
+
+    /**
+     * Extracts the project id from a work package.
+     * @param {*} workPackage Work package.
+     * @returns Project ID.
+     */
+    extractProjectID(projectURL) {
+        if (!projectURL) {
+            return null;
+        }
+        const n = projectURL.lastIndexOf('/');
+        const projectID = projectURL.substring(n + 1);
+        return parseInt(projectID);
+    }
+
+    /**
+     * Get category id from project.
+     * @param {*} project Project.
+     * @returns {number|null} Category ID (Project ID).
+     */
+    getCategoryIDFromProject(project) {
+        const parent = project?._links?.parent?.href
+
+        if (!parent) {
+            return project.id;
+        }
+
+        return this.extractProjectID(parent);
+
+        // TODO: Use ancestor collection?
+        // const ancestors = project._links.ancestors;
+
+        // if (!ancestors?.length || ancestors?.length === 0) {
+        //     return project["id"];
+        // }
+
+        // const projectURL = ancestors[ancestors.length - 1].href;
+        // const projectID = this.extractProjectID(projectURL);
+        // return projectID
+    }
+
+    processData() {
+        for (let categoryID of this.categories) {
+            const category = this.projects[categoryID] ?? this.apiClient.loadProject(categoryID);
+            const id = "openproject-category-" + categoryID;
+
+            let tymeCategory = Category.fromID(id) ?? Category.create(id);
+            tymeCategory.name = category.name;
+        }
+
+        for (let projectID in this.projects) {
+            const project = this.projects[projectID];
+
+            if (!project) {
+                continue;
+            }
+
+            this.createOrUpdateProject(project);
+        }
+
+        for (let workPackageID in this.workPackages) {
+            const workPackage = this.workPackages[workPackageID];
+
+            if (!workPackage) {
+                continue;
+            }
+
+            this.createOrUpdateTask(workPackage);
+        }
+    }
+
+    /**
+     * Creates or updates an existing project in Tyme.
+     * @param {*} project Project from OpenProject.
+     */
+    createOrUpdateProject(project) {
+        const id = "openproject-project-" + project.id;
```

Comment:
Just `openproject-` as a prefix would be sufficient for all entities.

Reply:
Makes sense, I've adjusted the prefix for all entities.